### PR TITLE
Feature/compare wishlist badges

### PR DESCRIPTION
**Pull request by vishal-7037**
### Short description and why it's useful

Displayed item's count badges on compare and wishlist icon at the header section

### Screenshots of visual changes before/after (if there are any)

2019_06_12_Selection_001

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [X] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [X] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [X] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: core/modules/compare/store/getters.ts [hunk 1]
```diff
@@ -5,7 +5,8 @@ import CompareState from '../types/CompareState'
 const getters: GetterTree<CompareState, RootState> = {
   isEmpty: (state) => state.items.length === 0,
   isOnCompare: (state) => (product) => state.items.find(p => p.sku === product.sku),
-  isCompareLoaded: state => state.loaded
+  isCompareLoaded: state => state.loaded,
+  totalProducts: (state) => state.items.length
```
Comment (patzick) [inline core/modules/compare/store/getters.ts]:
```suggestion
  getCompareProductsCount: (state) => state.items.length
```

File: core/modules/wishlist/components/WishlistButton.ts [hunk 2]
```diff
@@ -3,5 +3,10 @@ export const WishlistButton = {
     toggleWishlist () {
       this.$store.dispatch('ui/toggleWishlist')
     }
+  },
+  computed: {
+    wishlistItemsCount () {
```
Comment (patzick) [inline core/modules/wishlist/components/WishlistButton.ts]: you could use `...mapGetters` for that to not duplicate logic and then use `getWishlistItemsCount`
```js
...mapGetters('wishlist', ['getWishlistItemsCount'])
```

File: core/modules/wishlist/store/index.ts [hunk 3]
```diff
@@ -13,6 +13,7 @@ export const module: Module<WishlistState, RootState> = {
   actions,
   mutations,
   getters: {
-    isWishlistLoaded: state => state.loaded
+    isWishlistLoaded: state => state.loaded,
+    wishlistItemCount: (state) => state.items.length
```
Comment (patzick) [inline core/modules/wishlist/store/index.ts]:
```suggestion
    getWishlistItemsCount: (state) => state.items.length
```

File: core/compatibility/components/blocks/Header/CompareIcon.js [hunk 4]
```diff
@@ -7,6 +7,9 @@ export default {
     isActive () {
       // Computed Property renamed to 'isEmpty'
       return !this.isEmpty
+    },
+    getTotalProducts () {
```
Comment (patzick) [inline core/compatibility/components/blocks/Header/CompareIcon.js]: this should be removed as we have it in CompareButton.ts

File: core/compatibility/components/blocks/Header/WishlistIcon.js [hunk 5]
```diff
@@ -13,6 +13,9 @@ export default {
     // deprecated in this component
     isWishlistOpen () {
       return this.$store.state.ui.wishlist
+    },
+    getwishlistItemsCount () {
```
Comment (patzick) [inline core/compatibility/components/blocks/Header/WishlistIcon.js]: this getter should be done with
```js
    ...mapGetters('wishlist', ['getWishlistItemsCount']),
```

File: core/modules/compare/components/CompareButton.ts [hunk 6]
```diff
@@ -6,6 +6,9 @@ export const CompareButton = {
   computed: {
     isEmpty (): boolean {
       return this.$store.getters['compare/isEmpty']
+    },
+    totalProducts () {
```
Comment (patzick) [inline core/modules/compare/components/CompareButton.ts]: this should be mapped getter
```js
...mapGetters('compare', ['getCompareProductsCount']),
```

File: src/themes/default/components/core/blocks/Header/CompareIcon.vue [hunk 7]
```diff
@@ -1,6 +1,13 @@
 <template>
   <router-link :to="localizedRoute('/compare')" class="compare-icon no-underline inline-flex" v-if="isActive">
     <i class="material-icons">compare</i>
+    <span
+      class="compare-count absolute flex center-xs middle-xs border-box py0 px2 h6 lh16 weight-700 cl-white bg-cl-silver"
+      v-cloak
+      v-show="getTotalProducts"
```
Comment (patzick) [inline src/themes/default/components/core/blocks/Header/CompareIcon.vue]:
```suggestion
      v-show="getCompareProductsCount"
```

File: src/themes/default/components/core/blocks/Header/CompareIcon.vue [hunk 8]
```diff
@@ -1,6 +1,13 @@
...
+    >
+      {{ getTotalProducts }}
```
Comment (patzick) [inline src/themes/default/components/core/blocks/Header/CompareIcon.vue]:
```suggestion
      {{ getCompareProductsCount }}
```

File: src/themes/default/components/core/blocks/Header/WishlistIcon.vue [hunk 9]
```diff
@@ -1,18 +1,42 @@
...
-    class="inline-flex bg-cl-transparent brdr-none"
+    class="inline-flex bg-cl-transparent brdr-none relative"
     @click="toggleWishlistPanel"
     :aria-label="$t('Open wishlist')"
   >
     <i class="material-icons">favorite_border</i>
+    <span
+      class="whishlist-count absolute flex center-xs middle-xs border-box py0 px2 h6 lh16 weight-700 cl-white bg-cl-silver"
+      v-cloak
+      v-show="getwishlistItemsCount"
```
Comment (patzick) [inline src/themes/default/components/core/blocks/Header/WishlistIcon.vue]:
```suggestion
      v-show="getWishlistItemsCount"
```

File: src/themes/default/components/core/blocks/Header/WishlistIcon.vue [hunk 10]
```diff
@@ -1,18 +1,42 @@
...
+    >
+      {{ getWishListQty }}
```
Comment (patzick) [inline src/themes/default/components/core/blocks/Header/WishlistIcon.vue]:
```suggestion
      {{ getWishlistItemsCount }}
```

File: src/themes/default/components/core/blocks/Header/WishlistIcon.vue [hunk 11]
```diff
@@ -1,18 +1,42 @@
...
 <script>
 import WishlistIcon from '@vue-storefront/core/compatibility/components/blocks/Header/WishlistIcon'
+import wishlistMountedMixin from '@vue-storefront/core/modules/wishlist/mixins/wishlistMountedMixin'
+import { WishlistButton } from '@vue-storefront/core/modules/wishlist/components/WishlistButton'
 
 export default {
-  mixins: [WishlistIcon]
+  mixins: [WishlistIcon, wishlistMountedMixin, WishlistButton],
+  computed: {
+    getWishListQty () {
```
Comment (patzick) [inline src/themes/default/components/core/blocks/Header/WishlistIcon.vue]: we should remove this computed as we have `getWishlistItemsCount ` in WishlistButton mixin

**Comment by vishal-7037** [comment]
@patzick ,
I'll update soon and let you know.
Thanks

**Comment by vishal-7037** [comment]
@patzick ,

I've updated all suggested changes. Please review it and let me know.

Thanks.

**Comment by vishal-7037** [comment]
@patzick & @pkarw ,

I've updated all suggested changes. Please review it and let me know if any changes.

Thanks.

**Pull request merged**